### PR TITLE
Median in numpy frontend

### LINKED ISSUE #2999: median
#1529

### PR DESCRIPTION
**Pull request by 5hv5hvnk**
closes #2999

### CODE REVIEW

File: ivy/functional/frontends/numpy/statistics/median.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+):
+    if dtype:
+        a = ivy.astype(ivy.array(a), ivy.as_ivy_dtype(dtype))
+    ret = ivy.median(a, out=out)
+    if ivy.is_array(where):
+        ret = ivy.where(where, ret, ivy.default(out, ivy.zeros_like(ret)), out=out)
+    return ret
+
+
+medain.unsupported_dtypes = {"torch": ("float16",)}
```
Comment (Daniel4078) [inline ivy/functional/frontends/numpy/statistics/median.py:23]: what is "medain"? Please fix this, thank you.

File: ivy/functional/frontends/numpy/statistics/median.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import ivy
+
+
+def median(
```
Comment (Daniel4078) [inline ivy/functional/frontends/numpy/statistics/median.py:4]: This is the input arguments that you are supposed to recreated I suppose: numpy.median(a, axis=None, out=None, overwrite_input=False, keepdims=False)
Also, ivy.median is not implemented yet. So you may convert your existing GitHub issue to request for the function to be added to Ivy. Meanwhile, you can select another frontend function to work on from the ToDo list!

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Well, I have to say that I do not understand your implementation here and there are some problems to fix. Thank you.

File: ivy/functional/frontends/numpy/statistics/median.py (hunk 2 above)
Comment (Daniel4078) [reply ivy/functional/frontends/numpy/statistics/median.py:4]: Or you may try using the other existing ivy functions to implement this function instead.

**Comment by Daniel4078** [comment]
by the way, please write a test for it in this directory: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics while following the naming convention of Ivy.

File: ivy/functional/frontends/numpy/statistics/median.py (hunk 2 above)
Comment (5hv5hvnk) [reply ivy/functional/frontends/numpy/statistics/median.py:4]: @Daniel4078 where can I add the median function in ivy itself (which file)?

Comment (Daniel4078) [reply ivy/functional/frontends/numpy/statistics/median.py:4]: Firstly, you do not need to implement ivy.median yourself as it is someone else's job. Secondly, if it is implemented, I suppose it should be in ivy\functional\ivy\statistical.py.
But as I have said, you can just change your issue into a request of implementing such function and let other professional developers deal with the problem.

Comment (5hv5hvnk) [reply ivy/functional/frontends/numpy/statistics/median.py:4]: Alright, is there a function to flatten an ivy array?

**Comment by 5hv5hvnk** [comment]
> by the way, please write a test for it in this directory: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics while following the naming convention of Ivy.

Sure give me some time, If you can go through latest commit can you tell If the function seems fine to you

File: ivy/functional/frontends/numpy/statistics/median.py (hunk 2 above)
Comment (Daniel4078) [reply ivy/functional/frontends/numpy/statistics/median.py]: I found a flatten function in the helper functions for tests. I suppose you should not call the function, but you can use its code directly instead.
def flatten(*, ret):
    """Returns a flattened numpy version of the arrays in ret."""
    if not isinstance(ret, tuple):
        ret = (ret,)
    ret_idxs = ivy.nested_indices_where(ret, ivy.is_ivy_array)
    return ivy.multi_index_nest(ret, ret_idxs)

File: ivy/functional/frontends/numpy/statistics/median.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    if overwrite_input:
+        med = median_overwrite(a, axis, keepdims)
+    else:
+        med = _median(a, axis)
+
+    if out is not None:
+        out = med
+        return
+
+    return med
```
Comment (Daniel4078) [inline ivy/functional/frontends/numpy/statistics/median.py:51]: I found that you forget about the line that states the unsupported_dtypes. And you should have a blank line at the end of each file according to the formatting requirements.

**Comment by Daniel4078** [comment]
just a small tip, you can request a fresh review from a reviewer by click the refresh icon in the Reviewers section of the sidebar of the Conversation tab.

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
I suppose this median function should be placed in the average_and_variances.py file, please change that.

**Comment by Daniel4078** [comment]
You should only use ivy functions in your implementation instead of functions from other frameworks as the function is supposed to work without relying on other frameworks. Thank you.

**Comment by Daniel4078** [comment]
I just got an update on the policy of how to deal with this pull request's situation.
In a nutshell, the contributor can either implement the function as a composition, with a “#ToDo” comment added in the implementation, or they can add the “Next Release” label to the “Sub-Task” issue, and then choose another frontend function to work on.
Thank you.

**Comment by 5hv5hvnk** [comment]
@Daniel4078 I have replaced `numpy.flatten` with `.reshape(1,-1)` using ivy removing bumpy dependency in code

**Comment by Daniel4078** [comment]
I can only certainly decide if the current implementation has all the desired behaviors when you produce a test for it. But for now, I think it looks fine to me.

**Comment by Daniel4078** [comment]
Speaking about testing, here is a change about it:
The use of the  decorator handle_cmd_line_args has been changed a little. So now we're wrapping the  given decorator, instead of the test function. This is done so as to set the backend before assigning any values in the given decorator. We should now be using ivy.current_backend_str().valid_dtypes inside our helper functions, instead of ivy_np.valid_dtypes(i'm currently in the process of updating, this). Also, please don't pass in anything for as_variable, native_array, container, with_out, instance_method inside given, since that is being handled by the decorator, and it will only overwrite the values for these flags, which we don't want.

**Comment by 5hv5hvnk** [comment]
Hey sorry I got busy with my final submission can I continue working in this ?

**Comment by Daniel4078** [comment]
> Hey sorry I got busy with my final submission can I continue working in this ?

of course you can, so firstly can you fix the conflicting file so the automated checks can start? Thank you.

**Comment by 5hv5hvnk** [comment]
sure let me resolve conflicts and write the tests

**Pull request closed without merging**